src/views/templates/actions/components/CloneTemplateModal.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sCreate, K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { ButtonVariant, Form, FormGroup, TextInput } from '@patternfly/react-core';
+
+type CloneTemplateModalProps = {
+  isOpen: boolean;
+  obj: V1Template;
+  onClose: () => void;
+};
+
+const CloneTemplateModal: React.FC<CloneTemplateModalProps> = ({ isOpen, obj, onClose }) => {
+  const { t } = useKubevirtTranslation();
+  const [templateName, setTemplateName] = React.useState('');
+  const [templateProvider, setTemplateProvider] = React.useState('');
+
+  const onSubmit = async () => {
+    await k8sCreate<V1Template>({
+      model: TemplateModel,
+      data: {
+        apiVersion: obj?.apiVersion,
+        kind: TemplateModel.kind,
```

Comment:
- no need to change the `kind`, obj is a template, it should have the currect kind
- if you only want to change the provider, you should take `obj` and change only the provider
- you also need to change `template.kubevirt.io/type` label from `base` to `vm` because after clonning it's not a common template anymore
- do you need to change the `namespace` ?

to change only one element in an obj we have `immer` you can look how it's used in the wizard
another option is to use some `rfc6902` package that allow to patch an object, but I don't think we have one @glekner @upalatucci ?